src/mlpack/methods/reinforcement_learning/environment/continuous_mountain_car.hpp

```diff
@@ -120,24 +124,25 @@ class ContinuousMountainCar
   {
     // Calculate acceleration.
     double force = std::min(std::max(action.action[0], -1.0), 1.0);
-    nextState.Velocity() = state.Velocity() + 0.001 * force - 0.0025 *
+
+    // Update states.
+    nextState.Velocity() = state.Velocity() + force * power - 0.0025 *
         std::cos(3 * state.Position());
     nextState.Velocity() = std::min(
         std::max(nextState.Velocity(), velocityMin), velocityMax);
-
-    // Update states.
     nextState.Position() = state.Position() + nextState.Velocity();
     nextState.Position() = std::min(
         std::max(nextState.Position(), positionMin), positionMax);
-
     if (nextState.Position() == positionMin && nextState.Velocity() < 0)
-    {
       nextState.Velocity() = 0.0;
-    }
-    // If it is a terminal state, reward is 100.0
+
+    // Calculate reward
+    double reward = 0.0;
+    // If it is a terminal state, add a reward of 100.0
     if (IsTerminal(nextState))
-      return 100.0;
-    return -pow(action.action[0], 2)*0.1;
+      reward = 100.0;
+    reward -= std::pow(action.action[0], 2) * 0.1;
```

Comment:
Don't think this makes a huge difference, but I guess it's a good idea to get the same results as the gym env.